Adding tutorials for deployment on Heroku and Digital Ocean

malloxpb opened the pull request:
Fixes: #

Improves: Since it took me a lot of time figuring out the platforms I should use to deploy my Plone site as well as how to deploy it, I made this tutorial to help people save time.

Changes proposed in this pull request:

-Guide users to deploying Plone on Heroku step-by-step to get deeper understanding on how it works

-Add the deployment on Digital Ocean tutorial to guide new users deploy their first Plone app on VPS more easily.

-Give common errors people might get into while deploying their apps on Digital Ocean.
[preview.pdf](https://github.com/plone/documentation/files/938859/preview.pdf)

malloxpb committed:
Create a new page for the deploying on DO tutorial

malloxpb committed:
Add verification ssh instruction

malloxpb committed:
Add pictures for Mac and windows shell.

malloxpb committed:
Fix tab indentation

malloxpb committed:
Change doc name, add heroic deployment

malloxpb committed:
add more information to VPS deployment

malloxpb committed:
Remove unnecessary images, fix grammar mistakes

malloxpb committed:
Fix grammar mistakes and trailing space

malloxpb committed:
fix grammar mistake

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+=============================
+Deploy your Plone app to Heroku and Digital Ocean
+=============================
+
```
Not valid reST, please follow: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Deploying your Plone site on Heroku
+------------
+
```
Not valid reST, please follow: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Introduction
+------------
+
```
Wring header, please follow: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Setting up Heroku account and local environment
+^^^^^^^^^
+
```
Not valid reST, please follow: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Download and deploy Plone
+^^^^^^^^^
+
```
Not valid reST and wrong header, please follow: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+    git clone https://github.com/collective/training-sandbox
+    $ cd training-sandbox
+
```
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+    $ git remote -v
+        heroku	https://git.heroku.com/[your-app-name].git (fetch)
+        heroku	https://git.heroku.com/[your-app-name].git (push)
+
```
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+    git add .
+    $ git commit -m "your commit message"
+    $ git push heroku master
+
```
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Deploying your Plone app to Digital Ocean
+------------
+
```
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Setting up accounts and droplets
+^^^^^^^^^
+
```
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+.. image:: ../images/create_droplet.png
+    :align: center
+    :alt: Create droplet
+
```
Wrong location for pictures, they should go into the _static under /root of the repo, 
please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+.. image:: ../images/create_dropletos.png
+    :align: center
+    :alt: Choose droplet OS
+
```
Wrong location for pictures, they should go into the _static under /root of the repo,
please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+.. image:: ../images/create_dropletsize.png
+    :align: center
+    :alt: Choose droplet size
+
```
Wrong location for pictures, they should go into the _static under /root of the repo,
please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+.. note::
+    In case you want to deploy your app using a specific domain name, you can learn how to set it up at `Digital Ocean documentation <https://www.digitalocean.com/community/tutorials/how-to-set-up-a-host-name-with-digitalocean>`_
+
```
Typo . is missing

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Sign in to your Droplet 
+^^^^^^^^^
+
```
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
``````````diff
@@ -0,0 +1,226 @@
+
+Mac and Linux
+`````````
+
``````````
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
``````````diff
@@ -0,0 +1,226 @@
+
+Windows
+`````````
+
``````````
Please follow our style-guide: https://docs.plone.org/about/rst-styleguide.html

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+.. image:: ../images/winputty.png
+    :align: center
+    :alt: Windows putty setup
+
```
Wrong location for pictutres

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Deploy Plone to VPS
+^^^^^^^^^
+
```
Not valid reST and wrong header

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+    cd Plone-5.0.7-UnifiedInstaller
+    $ ./install.sh $OPTION
+
```
This is creating 'double' content, better use :ref: or :doc: for linking to the installation documentation

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+    cd /opt/plone/zinstance
+    $ sudo -u plone_buildout bin/buildout
+    $ ./bin/instance fg
+
```
Please follow our style-guide

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+
+Common errors
+^^^^^^^^^
+
```
Wrong header and not valid reST

svx commented on manage/deploying/production/heroku_do.rst:
``````````diff
@@ -0,0 +1,226 @@
+
+Cannot install lxml==3.5.0
+`````````
+
``````````
Not valid reST

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+      Installing instance.
+      Getting distribution for 'lxml==3.5.0'.
+    Error: Couldn't install: lxml 3.5.0
+
```
Formating is not valid

svx commented on manage/deploying/production/heroku_do.rst:
``````````diff
@@ -0,0 +1,226 @@
+
+The Installer does not generate all the Plone files
+`````````
+.. image:: ../images/errorDO.png
``````````
Not valid reST and wrong header

svx commented on manage/deploying/production/heroku_do.rst:
```diff
@@ -0,0 +1,226 @@
+    :alt: droplet installation error
+
+This error happens when the default installation path in Unified Installer has been modified.
+**How to fix**: run the Unified Installer again. But you need to leave the installation path as default. 
```
Using ..note:: would be better

svx commented on manage/deploying/production/index.rst:
```diff
@@ -13,5 +13,6 @@ Production
     restarts
     ubuntu_production
+    heroku_do
     securitybestpractices
```
Better use a filename which is clear indicating what it is about

svx commented:
Thanks for you first contribution and taking the time to working on that !

There are a couple of 'issues' :)

First, please make sure that you follow [our style-guide](https://docs.plone.org/about/rst-styleguide.html), currently the reST syntax is  not valid for headers, code-examples and so on.

Second you use the wrong header types  :)

Third, we have one place where we store docs, this is the *_static* directory in the root.
This is also written in our style-guide :)

One other thing is that you creating double content, because you copy already written content and place it under a new location, it is better to link or reference here, this is easier to maintain and also better for user, because with your solution, this content shows up  twice now, if people searching for it, which is confusing :)

Taking is further, we have already docs about Heroku, there are located in the repositories:

- https://github.com/plone/heroku-buildpack-plone
- https://github.com/plone/heroku-button-plone

Following our setup, I would prefer to check if your parts can added there as improvements and than, if we really want, we fetch the docs from there.

About the DO docs, IMHO, you find on DO self, a excellent documentation, how you create droplets use ssh and stuff like this.
If we add all this to our docs, this creates extra work for us, because we have to make sure that this is always is sync and up to date with DO.

Currently at PLOG [Plone Open Garden] people are busy with working on a new installer overview site, IMHO this should be the place where we may add instructions, our docs are more about Plone itself :)

Never the less, thanks for all your work, this was for sure not for nothing, we 'just' have to see how and where we use it :)

Thank you again !

stevepiercy commented:
In addition to the above suggestions, I found the content to be both unique and applicable to Plone from [line 151](https://github.com/nctl144/documentation/blame/c37cd8fb986793d9aa2a41c5268ea378706fc8f1/manage/deploying/production/heroku_do.rst#L151) onward. I doubt I would have figured out how to fix the lxml install issue.

malloxpb commented:
Hey guys,

Thank you so much for your comments on this PR. I really appreciate it :). I will try to work on fixing the content as soon as possible and come up with solutions on how to use this in the Plone documentation :)

malloxpb committed:
Fix indentation

malloxpb commented:
@svx After a few days working on this, I have deleted the Heroku part in this PR and I think I am going to contribute to the Heroku README files to make the deployment tutorial easier (if it is a good idea in your opinion). 
About the indentation and the ReST styling for the documentation, I have fixed the headings, images locations, etc.. However, I can see that the default branch has changed to 5.1. Should I make another pull request to that branch and close this PR?
